src/Catel.MVVM/Collections/ObservableDictionary.cs

```diff
@@ -0,0 +1,682 @@
+        {
+            Add(item.Key, item.Value);
+        }
+
+        /// <inheritdoc cref="IDictionary.Clear"/>
+        public void Clear()
+        {
+            if (Count <= 0)
+            {
+                return;
+            }
+
+            _collection.Clear();
+            _version++;
+
+            OnPropertyChanged();
+            OnCollectionChanged();
+        }
+
+        /// <inheritdoc cref="ICollection{T}.Contains"/>
+        public bool Contains(KeyValuePair<TKey, TValue> item)
+        {
+            return _collection.Contains(item);
+        }
+
+        /// <inheritdoc cref="IDictionary{TKey, TValue}.ContainsKey"/>
+        public bool ContainsKey(TKey key)
+        {
+            return _collection.Contains(key);
+        }
+
+        /// <inheritdoc cref="ICollection{T}.CopyTo"/>
+        public void CopyTo(KeyValuePair<TKey, TValue>[] array, int arrayIndex)
+        {
+            Argument.IsNotNull(nameof(array), array);
+            Argument.IsNotOutOfRange(nameof(arrayIndex), arrayIndex, 0, array.Length);
+
+            if (array.Rank != 1)
+            {
+                throw Log.ErrorAndCreateException<ArgumentException>("Multidimensional arrays are not supported.");
+            }
+
+            if (array.GetLowerBound(0) != 0)
+            {
+                throw Log.ErrorAndCreateException<ArgumentException>("Array is non-zero lower bound.");
+            }
+
+            if (array.Length - arrayIndex < _collection.Count)
+            {
+                throw Log.ErrorAndCreateException<ArgumentException>("Supplied array was too small.");
+            }
+
+            foreach (KeyValuePair<TKey, TValue> keyValuePair in _collection)
+            {
+                array[arrayIndex++] = new KeyValuePair<TKey, TValue>(keyValuePair.Key, keyValuePair.Value);
+            }
+        }
+
+        /// <inheritdoc cref="IEnumerable{T}.GetEnumerator"/>
+        public IEnumerator<KeyValuePair<TKey, TValue>> GetEnumerator()
+        {
+            return new Enumerator(this);
+        }
+
+        /// <inheritdoc cref="IDictionary{TKey, TValue}.Remove(TKey)"/>
+        public bool Remove(TKey key)
+        {
+            if (!TryGetValue(key, out TValue value))
+            {
+                return false;
+            }
+
+            KeyValuePair<TKey, TValue> item = new KeyValuePair<TKey, TValue>(key, value);
+            int index = _collection.IndexOf(item);
+
+            if (!_collection.Remove(key))
+            {
+                return false;
+            }
+
+            OnCollectionChanged(NotifyCollectionChangedAction.Remove, item, index);
+
+            _version++;
+
+            return true;
+        }
+
+        /// <inheritdoc cref="IDictionary{TKey, TValue}.TryGetValue"/>
+        public bool TryGetValue(TKey key, out TValue value)
+        {
+            bool result = _collection.Contains(key);
+
+            value = result ? _collection[key].Value : default;
```

Comment:
TryGetValue on _collection itself or does a KeyedCollection not support that?

Reply:
Good catch on this one, actually had to do some digging. TryGetValue only exists on .NET Core for KeyedCollection so I adjusted the code with a compiler condition to use that method for .NET Core and the existing code for .NET Framework.